### PR TITLE
Feature/135: Implement graphs on homepage

### PR DESCRIPTION
**Pull request by emilie-martin**
image

### CODE REVIEW

**Review by nasaku898 — APPROVED** [review]

File: bike-erp/backend/src/main/routes/component_routes.ts [hunk 1]
```diff
@@ -42,7 +42,19 @@ router.post("/addComponent", authenticateToken, verifyRole([Role.ADMIN, Role.MAN
...
+    .getComponentsByLocation(loc)
+    .then((response) => {
+      res.json(response);
+    })
+    .catch((error) => {
+      res.status(error.status).send(error.message);
+    });
+});
+
+router.get("/:component_id", authenticateToken, verifyRole([Role.ADMIN, Role.MANAGER, Role.EMPLOYEE]), (req, res) => {
```
Comment (Manraj45) [inline bike-erp/backend/src/main/routes/component_routes.ts:57]: You need to add the following here:
image
When you try to order a bike as a customer you get unauthorized user:
image

Comment (Manraj45) [reply bike-erp/backend/src/main/routes/component_routes.ts:57]: router.get("/:component_id", authenticateToken, verifyRole([Role.ADMIN, Role.MANAGER, Role.EMPLOYEE, Role.CUSTOMER]), (req, res) => {

**Review by Manraj45 — CHANGES_REQUESTED** [review]
Small bug where customer can't order bikes

File: bike-erp/backend/src/main/routes/component_routes.ts (hunk 1 above)
Comment (emilie-martin) [reply bike-erp/backend/src/main/routes/component_routes.ts]: good catch. don't know how that slipped it should be fixed

**Review by Manraj45 — APPROVED** [review]
Good Job!!

**Pull request merged**